### PR TITLE
[DEV-804] - Add GitBook embed parsing

### PR DESCRIPTION
**Pull request by kin0992**
#### List of Changes

- Add new `embed` schema
- Change `RenderingComponents`

#### Motivation and Context

Add the parsing for kind of gitbook embed blocks

#### How Has This Been Tested?

Unit Test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/gitbook-docs/src/__tests__/parseContent.test.ts [hunk 1]
```diff
@@ -255,4 +255,20 @@ describe('parseContent', () => {
       ]),
     ]);
   });
+
+  it.only('should parse embed', () => {
```
Comment (kin0992) [inline packages/gitbook-docs/src/__tests__/parseContent.test.ts:259]:
```suggestion
  it('should parse embed', () => {
```

File: packages/gitbook-docs/src/__tests__/parseContent.test.ts [hunk 2]
```diff
@@ -255,4 +255,20 @@ describe('parseContent', () => {
...
+      '{% endembed %}';
+    expect(parseContent(embed, config)).toStrictEqual([
+      new Markdoc.Tag(
+        'Embed',
+        {
+          url: 'https://www.pagopa.it/',
+        },
+        [new Markdoc.Tag('Paragraph', {}, ['This is a caption'])]
+      ),
+    ]);
```
Comment (datalek) [inline packages/gitbook-docs/src/__tests__/parseContent.test.ts:272]: Can you add a test that covers the [self-closed tag scenario](https://github.com/pagopa/devportal-docs/blob/docs/from-gitbook/docs/v7eFccIIRcGfOIG6IHkT/esperienza-per-il-cittadino/app-io/README.md?plain=1#L22)?

**Review by datalek — CHANGES_REQUESTED** [review]

File: packages/gitbook-docs/src/__tests__/parseContent.test.ts [hunk 3]
```diff
@@ -278,4 +278,33 @@ describe('parseContent', () => {
...
+
+    expect(
+      parseContent('{% embed url="https://www.pagopa.it/" %}\nhi', config)
+    ).toStrictEqual([
+      new Markdoc.Tag(
+        'Embed',
+        {
+          url: 'https://www.pagopa.it/',
+        },
+        []
```
Comment (datalek) [inline packages/gitbook-docs/src/__tests__/parseContent.test.ts:305]: You can omit the `[]` from constructor.

**Review by datalek — APPROVED** [review]
👓

**Pull request merged**